Sources/MapboxMaps/Foundation/MapView+Attribution.swift

```diff
@@ -0,0 +1,64 @@
+@_implementationOnly import MapboxCommon_Private
+
+@available(iOSApplicationExtension, unavailable)
+extension MapView: AttributionDialogManagerDelegate {
+    func viewControllerForPresenting(_ attributionDialogManager: AttributionDialogManager) -> UIViewController {
+        // Consider delegating out to a delegate
+        return parentViewController!
+    }
+
+    func attributionDialogManager(_ attributionDialogManager: AttributionDialogManager, didTriggerActionFor attribution: Attribution) {
+        let url: URL
+
+        if attribution.isFeedbackURL {
+            url = mapboxFeedbackURL()
+        } else {
+            url = attribution.url
+        }
+
+        Log.debug(forMessage: "Open url: \(url))", category: "Attribution")
+
+        UIApplication.shared.open(url)
+    }
+
+    internal func mapboxFeedbackURL() -> URL {
+        let cameraState = self.cameraState
+
+        var components = URLComponents(string: "https://apps.mapbox.com/feedback/")!
+        components.fragment = String(format: "/%.5f/%.5f/%.2f/%.1f/%i",
+                                     cameraState.center.longitude,
+                                     cameraState.center.latitude,
+                                     cameraState.zoom,
+                                     cameraState.bearing,
+                                     Int(round(cameraState.pitch)))
+
+        let applicationBundleId = Bundle.main.bundleIdentifier // com.apple.dt.xctest.tool during testing
+        let referrerQueryItem = URLQueryItem(name: "referrer", value: applicationBundleId)
+
+        var queryItems = [referrerQueryItem]
+
+        if let styleURIString = mapboxMap.style.uri?.rawValue,
+           let styleURL = URL(string: styleURIString),
+           styleURL.scheme == "mapbox",
+           styleURL.host == "styles" {
+            let pathComponents = styleURL.pathComponents
+            let sdkVersion = Bundle.mapboxMapsMetadata.version
+
+            var accessToken = resourceOptions?.accessToken ?? "unknown"
+            accessToken = accessToken.isEmpty ? "unknown" : accessToken
```

Comment:
in the current implementation, sdkVersion and accessToken are omitted if style.uri is nil. Is that intentional, or should we include sdkVersion and accessToken even if style.uri is nil?

Reply:
Nice catch, I think we should stick with current implementation

Reply:
This code block actually has a if let at the top that is checking to see if style uri is not nil